src/xibo-interactive-control.js

```diff
@@ -493,8 +519,75 @@ window.xiboIC = (function() {
         return _lib[widgetId][name].apply(null, args);
       }
     },
+
+    // Realtime data
+    /**
+     * Get realtime data from the player. Called from the widget.
+     * @param {string} dataSetId The id of the dataset
+     * @param {Object[]} [options] - Request options
+     * @param {callback} [options.done]
+     * @param {callback} [options.error]
+     */
+    getData(dataSetId, {done, error} = {}){
+      _lib.makeRequest(
+        '/realtime?dataSetId='+dataSetId,
+        {
+          type: 'GET',
+          done: done,
+          error: error,
+        },
+      );
+    },
+
+    /**
+     * Set the realtime into the player. Called from Data Connector.
+     * @param {string} dataSetId The id of the dataset
+     * @param {Object[]} data The data for the dataset as JSON array of objects
+     * @param {Object} options - Request options
+     * @param {callback} options.done Optional
+     * @param {callback} options.error Optional
+     */
+    setData(dataSetId, data, {done, error} = {}){
+      _lib.makeRequest(
+        '/realtime?dataSetId='+dataSetId,
+        {
+          type: 'POST',
+          data: data,
+          done: done,
+          error: error,
+        },
+      );
+    },
+
+    /**
+     * Notify main application that we have new data. Called from
+     * @param {string} dataSetId - The id of the dataset
+     * @param {string} widgetId - Optional. Widget id to notify. If omitted, all widgets will be notified.
+     */
+    notifyHost(dataSetId, widgetId) {
+      _IPC.sendData(
```

Comment:
This needs to cater for players which do not use IPC - i.e. Windows will use a webview binding.

I think we said that this would check for a `notifyHost` function declared in the window, and if it finds one it would call it? Each player can then implement the function to do what they need?

Reply:
You are right on that.

This is a pending change that I have to do.